### PR TITLE
1089 how expose prom route

### PR DESCRIPTION
**Pull request by leifmadsen**
- Add documentation about exposing routes
- Fix issues with wrapping

### CODE REVIEW

**Comment by leifmadsen** [comment]
[Add instructions to expose Prometheus route in STF documentation](https://trello.com/c/RWAvjm4O/1089-add-instructions-to-expose-prometheus-route-in-stf-documentation)

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// The `context` attribute enables module reuse. Every module's ID includes
+// {context}, which ensures that the module has a unique ID even if it is
+// reused multiple times in a guide.
+//
+// Start the title with a verb, such as Creating or Create. See also
+// _Wording of headings_ in _The IBM Style Guide_.
+[id="exposing-routes_{context}"]
+= Creating a route in {OpenShift}
+
+In {OpenShift} ingress traffic for applications is done by exposed services and accessing the application via a route. For more information about Configuring ingress cluster traffic, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].
```
Comment (csibbitt) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:26]: I find the first sentence hard to parse. How about something a bit simpler, like "In {OpenShift}, applications can be exposed to the external network via a route."

**Review by csibbitt — APPROVED** [review]

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc (hunk 1 above)
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:26]:
```suggestion
In {OpenShift}, you can expose applications to the external network via a route. For more information, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].
```

I agree with @csibbitt and I like his suggested edit.

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+By default in {Project} routes are not exposed in order to limit the attack surface of {ProjectShort} deployments. In order to access some services deployed in {ProjectShort} the services in {OpenShiftShort} must be exposed for access.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:28]:
```suggestion
Routes are not exposed by default in {Project} to limit the attack surface of {ProjectShort} deployments. To access some services deployed in {ProjectShort}, you must expose the services in {OpenShiftShort} for access.
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+// Start the title with a verb, such as Creating or Create. See also
+// _Wording of headings_ in _The IBM Style Guide_.
+[id="exposing-routes_{context}"]
+= Creating a route in {OpenShift}
+
+In {OpenShift} ingress traffic for applications is done by exposed services and accessing the application via a route. For more information about Configuring ingress cluster traffic, see https://docs.openshift.com/container-platform/{SupportedOpenShiftVersion}/networking/configuring_ingress_cluster_traffic/overview-traffic.html[Configuring ingress cluster traffic].
+
+Routes are not exposed by default in {Project} to limit the attack surface of {ProjectShort} deployments. To access some services deployed in {ProjectShort}, you must expose the services in {OpenShiftShort} for access.
+
+The way services are exposed can be environment specific, but a common service to expose in {ProjectShort} is Prometheus.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:30]:
```suggestion
A common service to expose in {ProjectShort} is Prometheus, as shown in the following example:
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+. Log in to {OpenShift}.
+. Change to the `service-telemetry` namespace:
++
+[source,bash]
+----
+$ oc project service-telemetry
+----
+
+. List the available services in the `service-telemetry` project
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:42]:
```suggestion
. List the available services in the `service-telemetry` project:
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+default-cloud1-ceil-meter-smartgateway   ClusterIP   172.30.114.195   <none>        8081/TCP                                        93m
+default-cloud1-coll-meter-smartgateway   ClusterIP   172.30.133.180   <none>        8081/TCP                                        93m
+default-interconnect                     ClusterIP   172.30.3.241     <none>        5672/TCP,8672/TCP,55671/TCP,5671/TCP,5673/TCP   93m
+ibm-auditlogging-operator-metrics        ClusterIP   172.30.216.249   <none>        8383/TCP,8686/TCP                               11h
+prometheus-operated                      ClusterIP   None             <none>        9090/TCP                                        93m
+service-telemetry-operator-metrics       ClusterIP   172.30.11.66     <none>        8383/TCP,8686/TCP                               11h
+smart-gateway-operator-metrics           ClusterIP   172.30.145.199   <none>        8383/TCP,8686/TCP                               11h
+----
+
+. Note the port and service name to expose as a route to allow for external access, for example service `prometheus-operated` and port `9090`
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:62]:
```suggestion
. Take note of the port and service name to expose as a route, for example, service `prometheus-operated` and port `9090`.
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+. Expose the `prometheus-operated` service as an edge route and redirect insecure traffic to the secure endpoint of port `9090`
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:64]:
```suggestion
. Expose the `prometheus-operated` service as an edge route and redirect insecure traffic to the secure endpoint of port `9090`:
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+[source,bash,options="nowrap",subs="+quotes"]
+----
+$ oc create route edge metrics-store --service=prometheus-operated --insecure-policy="Redirect" --port=9090
+----
++
+----
+route.route.openshift.io/metrics-store created
+----
+
+. To verify and find the exposed external DNS for the route use the `oc get route` command
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:75]:
```suggestion
. To verify and find the exposed external DNS for the route, use the `oc get route` command:
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+[source,bash]
+----
+$ oc get route metrics-store -ogo-template='{{.spec.host}}'
+----
++
+----
+metrics-store-service-telemetry.apps.infra.watch
+----
+
+. The `prometheus-operated` service will now be avalable at the exposed DNS address, e.g. https://metrics-store-service-telemetry.apps.infra.watch
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc:86]:
```suggestion
+
The `prometheus-operated` service is now available at the exposed DNS address, for example, https://metrics-store-service-telemetry.apps.infra.watch
```

File: doc-Service-Telemetry-Framework/master.adoc [hunk 9]
```diff
@@ -1,4 +1,4 @@
-= Service Telemetry Framework {ProjectCurrentVersion}
+= Service Telemetry Framework
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/master.adoc:1]:
```suggestion
= Service Telemetry Framework {ProductVersion}
```

Can we include the version here so that the titles match with Pantheon titles?

**Review by JoanneOFlynn2018 — CHANGES_REQUESTED** [review]
Some very minor points for you to look over

File: doc-Service-Telemetry-Framework/master.adoc (hunk 9 above)
Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/master.adoc:1]: Unfortunately this doesn't work because the `stf-attributes.yml` file is included after that part of the `master.adoc`. I tried to move it further up, but it fails. We'll probably just have to manually adjust that unfortunately.

Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/master.adoc:1]:
```suggestion
= Service Telemetry Framework 1.2
```

File: doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc (hunk 1 above)
Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc]: Thanks! I wrote this really fast and did not review like I would normally do and instead relied on our awesome team to fix it :)

Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc]: Will review tomorrow thanks for picking this out. As noted earlier I didn't do my normal 2nd and 3rd readings and just brain dumped.

Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_exposing-routes.adoc]: Ah sweet Joanne already fixed this :)

**Comment by leifmadsen** [comment]
@JoanneOFlynn2018 I think this is ready to go now.

Thanks everyone!

**Review by JoanneOFlynn2018 — APPROVED** [review]
